security_and_privacy_self_review.md

```diff
@@ -0,0 +1,114 @@
+
+## Authors
+
+- Takumi Fujimoto [@takumif](https://github.com/takumif) |
+  [<takumif@google.com>](mailto:takumif@google.com)
+
+## 2.1 What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+A website would be able to query (get events fired containing a boolean) whether
+the user has devices on the local network that support being a receiver for
+site-initiated mirroring (e.g a Chromecast device). Note that it could already
+query for devices that support existing uses of Presentation API (launching a
+web app by URL), and those sets of devices overlap for the most part.
+
+This is needed for a website to be able to tell when it should show UI to allow
+users to initiate site-initiated mirroring -- it wouldn't be desirable for the
+user to click through just to find that they have no compatible devices.
+
+## 2.2 Do features in your specification expose the minimum amount of information necessary to enable their intended uses?
+
+Yes. We don't expose the number or the name of the available devices.
+
+## 2.3 How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?
+
+Existing Presentation API reduces PII exposure by showing the available list of
+devices (device names could contain people's names) in native UI and not
+exposing them to the webiste. Site-initiated mirroring, which extends the
+Presentation API, uses the same mechanism.
+
+## 2.4 How do the features in your specification deal with sensitive information?
+
+See section 2.3 above.
+
+## 2.5 Do the features in your specification introduce new state for an origin that persists across browsing sessions?
+
+No.
+
+## 2.6 Do the features in your specification expose information about the underlying platform to origins?
+
+No.
+
+## 2.7 Does this specification allow an origin to send data to the underlying platform?
+
+No.
+
+## 2.8 Do features in this specification enable access to device sensors?
+
+No.
+
+## 2.9 Do features in this specification enable new script execution/loading mechanisms?
+
+No.
+
+## 2.10 Do features in this specification allow an origin to access other devices?
+
+Yes, the feature allows an origin to choose to mirror its top-level browsing
+context to a device on the local network chosen by the user in a native device
+picker UI.
+
+## 2.11 Do features in this specification allow an origin some measure of control over a user agent’s native UI?
+
+Yes, the proposed feature uses an existing flow of `PresentationRequest#start()`
+triggering a native device picker UI, which can only be triggered with user
+activation.
+
+## 2.12 What temporary identifiers do the features in this specification create or expose to the web?
+
+Starting a mirroring session creates a `PresentationConnection` object with an
+ID, but the ID can be obtained only at creation and cannot be used to
```

Comment:
You may want to reference the part of the Presentation API spec that describes these IDs to clarify that they are not tied to users or browsing state.